### ISSUE TITLE
Object arrays of complex arrays can't be multiplied by numpy floats (migrated from Trac #1778)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3326]
Original ticket http://projects.scipy.org/numpy/ticket/1778
Reported 2011-03-22 by atmention:inducer, assigned to unknown.

```
>>> import numpy as np
>>> z = np.array([1j])
>>> zz = np.zeros((2,), dtype=object)
>>> zz.fill(z)
>>> zz
array([[ 0.+1.j], [ 0.+1.j]], dtype=object)
>>> zz * np.float64(5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for *: 'numpy.ndarray' and 'numpy.float64'
>>> zz * 5.
array([[ 0.+5.j], [ 0.+5.j]], dtype=object)
>>> 
```

In my humble estimation, that TypeError shouldn't be happening.

**Comment by thouis** [comment]
Comment in Trac by atmention:inducer, 2011-03-22

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-04-20

This looks to be fixed in current devel 416e1abb102ab3fd5fe266853109c370e74070cf. I don't know when the actual fixed happened. We should add a regression test and close the ticket.
